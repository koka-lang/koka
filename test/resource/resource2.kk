effect heap<s::S> { 
}

val heap = handler<heap<_s>> { 
}

fun foo() {
  using heap
  42
}

fun main() {
  println(foo())
}
