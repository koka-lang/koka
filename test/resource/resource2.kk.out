42
.tag-heap : string
foo : () -> int
heap : forall<s,a,e> (action : () -> <heap<s>|e> a) -> e a
main : () -> console ()